autorest/codegen/models/property.py

```diff
@@ -32,26 +32,36 @@ def __init__(
         self.required: bool = yaml_data.get("required", False)
         self.readonly: bool = yaml_data.get("readOnly", False)
         self.is_discriminator: bool = yaml_data.get("isDiscriminator", False)
-        # this bool doesn't consider you to be constant if you are a discriminator
-        self.constant: bool = isinstance(self.schema, ConstantSchema) and not self.is_discriminator
-
         if description:
             self.description = description
         else:
             self.description = yaml_data["language"]["python"]["description"]
 
-        validation_map: Dict[str, Union[bool, int, str]] = {}
+        self.client_default_value = client_default_value
+
+    @property
+    def constant(self) -> bool:
```

Comment:
decided to make `constant` and `validation_map` readonly properties to unclutter the main`__init__` and because we don't set them later